### PR TITLE
refactor: app actor cleanup

### PR DESCRIPTION
**Pull request by subotic**
Issue Number: DEV-

### Basic Requirements

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### PR Type

What kind of change does this PR introduce?

- [ ] Bugfix: represents bug fixes
- [x] Refactor: represents production code refactoring
- [ ] Feature: represents a new feature
- [ ] Documentation: documentation changes (no production code change)
- [ ] Chore: maintenance tasks (no production code change)
- [ ] Style: styles updates (no production code change)
- [ ] Test: all about tests: adding, refactoring tests (no production code change)
- [ ] Other... Please describe:

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### Does this PR change client-test-data?

- [ ] Yes (don't forget to update the JS-LIB team about the change)
- [ ] No

### CODE REVIEW

**Comment by subotic** [comment]
Just some minor cleanup making the code (hopefully) simpler to read

File: webapi/src/main/scala/org/knora/webapi/core/AppServer.scala [hunk 1]
```diff
@@ -97,19 +95,15 @@ final case class AppServer(
...
+      _ <- iiifs
+             .getStatus()
+             .flatMap {
+               case IIIFServiceStatusOK =>
+                 ZIO.logInfo("IIIF service running")
+               case IIIFServiceStatusNOK =>
+                 ZIO.logError("IIIF service not running") *> ZIO.die(new Exception("IIIF service not running"))
+             }
+             .when(requiresIIIFService)
       _ <- state.set(AppState.IIIFServiceReady)
```
Comment (BalduinLandolt) [inline webapi/src/main/scala/org/knora/webapi/core/AppServer.scala:107]: I'm just wondering if it's actually reasonable to set the AppState to IIIF stuff, if we don't require the IIIF service. So maybe this method could be rewritten to:

```scala
private def checkIIIFService(requiresIIIFService: Boolean): ZIO[Any, Nothing, Unit] =
  if (requiresIIIFService) {
    for {
      _      <- state.set(AppState.WaitingForIIIFService)
      status <- iiifs.getStatus()
      _      <- status match {
        case CacheServiceStatusNOK =>
          ZIO.logError("Cache service not running.") *> 
            ZIO.die(new Exception("Cache service not running."))
        case CacheServiceStatusOK =>
          ZIO.unit
      }
      _      <- state.set(AppState.IIIFServiceReady)
    } yield ()
  } else ZIO.unit
```

optionally, you could also use ZIO.when to test the bool condition. And even ZIO.whenCase to check the status

**Review by BalduinLandolt — APPROVED** [review]
note my comment on IIIF: not sure if we should use the stati WaitingForIIIF and IIIFReady in instances where no IIIF service is required

**Review by mpro7 — COMMENTED** [review]
Why the conditions (`appConfig.allowReloadOverHttp` and `requiresIIIFService`) were removed?

**Review by irinaschubert — APPROVED** [review]

File: webapi/src/main/scala/org/knora/webapi/core/AppServer.scala (hunk 1 above)
Comment (subotic) [reply webapi/src/main/scala/org/knora/webapi/core/AppServer.scala:107]: yes, good point. With ZIO I'm actually not sure if we need these states in such a detail at all, or if something waiting for startup is not good enough. Now that we have logging that works, we should quickly figure out where the startup is stuck. So I will merge it as is, and we can create a follow-up issue after we discuss it.

**Comment by subotic** [comment]
> Why the conditions (`appConfig.allowReloadOverHttp` and `requiresIIIFService`) were removed?

they are now checked inside `when`

**Pull request merged**